packages/trace-viewer/src/sw/main.ts

```diff
@@ -95,6 +95,26 @@ async function doFetch(event: FetchEvent): Promise<Response> {
 
     const traceUrl = url.searchParams.get('trace');
 
+    if (relativePath.startsWith('/snapshot/')) {
+      const { snapshotServer } = loadedTraces.get(traceUrl!) || {};
+      if (!snapshotServer)
+        return new Response(null, { status: 404 });
+      const response = snapshotServer.serveSnapshot(relativePath, url.searchParams, url.href);
+      if (isDeployedAsHttps)
+        response.headers.set('Content-Security-Policy', 'upgrade-insecure-requests');
+      return response;
+    }
+
+    if (relativePath.startsWith('/closest-screenshot/')) {
+      const { snapshotServer } = loadedTraces.get(traceUrl!) || {};
+      if (!snapshotServer)
+        return new Response(null, { status: 404 });
+      return snapshotServer.serveClosestScreenshot(relativePath, url.searchParams);
+    }
+
+    if (!client)
```

Comment:
Shouldn't this be at line 80? How come there is no client for a request?

Reply:
I'm not sure what's the exact reason, but in my testing it appeared that requests from within the iframe don't resolve to a client.

Reply:
We've figured out some more by now:

- some requests have `event.resultingClientId` instead of `event.clientId`
- some requests have `event.clientId`, but the client isn't yet available in `self.clients`

I've updated the PR to reflect this.